deployment/hasura/migrations/Aerie/20_patch_activity_goal_invocation_id/up.sql

```diff
@@ -0,0 +1,824 @@
+           source_scheduling_goal_id, source_scheduling_goal_invocation_id, created_at, start_offset, type, arguments, metadata, anchor_id, anchored_to_start
+    from diff_diff dd
+    join merlin.activity_directive ad
+      on dd.activity_id = ad.id
+    where ad.plan_id = plan_id_receiving
+      and (dd.change_type_supplying = 'modify' and dd.change_type_receiving = 'modify'))
+  insert into merlin.merge_staging_area (
+    merge_request_id, activity_id, name, tags, source_scheduling_goal_id, source_scheduling_goal_invocation_id,
+    created_at, created_by, last_modified_by,
+    start_offset, type, arguments, metadata, anchor_id, anchored_to_start, change_type)
+    select _merge_request_id, ad.id, ad.name, tags,  ad.source_scheduling_goal_id, ad.source_scheduling_goal_invocation_id,
+           ad.created_at, ad.created_by, ad.last_modified_by, ad.start_offset, ad.type, ad.arguments, ad.metadata,
+           ad.anchor_id, ad.anchored_to_start, 'none'
+    from false_modify fm
+    left join merlin.activity_directive ad
+      on (ad.plan_id, ad.id) = (plan_id_receiving, fm.activity_id);
+
+  -- 'modify' against 'delete' and inequal 'modify' against 'modify' goes into conflict table (aka everything left in diff_diff)
+  insert into merlin.conflicting_activities (merge_request_id, activity_id, change_type_supplying, change_type_receiving)
+  select begin_merge._merge_request_id, activity_id, change_type_supplying, change_type_receiving
+  from (select begin_merge._merge_request_id, activity_id
+        from diff_diff
+        except
+        select msa.merge_request_id, activity_id
+        from merlin.merge_staging_area msa) a
+  join diff_diff using (activity_id);
+
+  -- Fail if there are no differences between the snapshot and the plan getting merged
+  validate_non_no_op_status := null;
+  select change_type_receiving
+  from merlin.conflicting_activities
+  where merge_request_id = _merge_request_id
+  limit 1
+  into validate_non_no_op_status;
+
+  if validate_non_no_op_status is null then
+    select change_type
+    from merlin.merge_staging_area msa
+    where merge_request_id = _merge_request_id
+    and msa.change_type != 'none'
+    limit 1
+    into validate_non_no_op_status;
+
+    if validate_non_no_op_status is null then
+      raise exception 'Cannot begin merge. The contents of the two plans are identical.';
+    end if;
+  end if;
+
+
+  -- clean up
+  drop table supplying_diff;
+  drop table receiving_diff;
+  drop table diff_diff;
+end
+$$;
+
+alter table merlin.plan_snapshot_activities
+  add column source_scheduling_goal_invocation_id integer;
+
+alter table merlin.merge_staging_area
+  add column source_scheduling_goal_invocation_id integer;
+
+comment on column merlin.merge_staging_area.source_scheduling_goal_id is e''
+  'The invocation of the scheduling goal that generated this activity directive to be committed.';
```

Comment:
Typo in the column name:
```suggestion
comment on column merlin.merge_staging_area.source_scheduling_goal_invocation_id is e''
  'The invocation of the scheduling goal that generated this activity directive to be committed.';
```

Reply:
...But I see that that was caused by the same typo in `merge_staging_area`